core/unix/src/TUnixSystem.cxx

```diff
@@ -4684,23 +4684,34 @@ static const char *DynamicPath(const char *newpath = nullptr, Bool_t reset = kFA
       dynpath_syspart = "/usr/local/lib:/usr/X11R6/lib:/usr/lib:/lib:";
       dynpath_syspart += "/lib/x86_64-linux-gnu:/usr/local/lib64:/usr/lib64:/lib64:";
    #else
-      // trick to get the system search path
-      std::string cmd("LD_DEBUG=libs LD_PRELOAD=DOESNOTEXIST ls 2>&1");
-      FILE *pf = popen(cmd.c_str (), "r");
-      std::string result = "";
-      char buffer[128];
-      while (!feof(pf)) {
-         if (fgets(buffer, 128, pf) != nullptr)
-            result += buffer;
-      }
-      pclose(pf);
-      std::size_t from = result.find("search path=", result.find("(LD_LIBRARY_PATH)"));
-      std::size_t to = result.find("(system search path)");
-      if (from != std::string::npos && to != std::string::npos) {
-         from += 12;
-         std::string sys_path = result.substr(from, to-from);
-         sys_path.erase(std::remove_if(sys_path.begin(), sys_path.end(), isspace), sys_path.end());
-         dynpath_syspart = sys_path.c_str();
+      // Obtain the system search path ...
+
+      // First of all, let's see if an outside entity gave us the system path.
```

Comment:
Maybe this setting could be documented either in the doxygen section of this function, or in a dedicated section of "Optimizations"

Reply:
Adding a meaningful doxygen statement would require a few lines with some technical detail. Something along the lines of 
```
/// The runtime cost (syscalls) of this function can be shortened by injecting
/// a pre-calculated result for the system library search path in form of
/// `export ROOT_LDSYSPATH=export ROOT_LDSYSPATH=$(LD_DEBUG=libs LD_PRELOAD=DOESNOTEXIST ls /tmp/DOESNOTEXIST 2>&1 | grep -m 1 "system search path" | sed 's/.*=//g' | awk '//{print $1}')`
/// This might be useful in scenarios, where ROOT is instantiated many times.
```
 
I can add it if you like. But please let me know if this statement is ok or if you have something else in mind, in order converge more quickly.

Reply:
LGTM, thanks.